### PR TITLE
Refactoring RAM dirs

### PR DESCRIPTION
**Pull request by b-scholz**
Move utility RAM headers into the src/ram/utility directory.
This PR is related to issue #462.

### CODE REVIEW

File: src/interpreter/Node.h [hunk 1]
```diff
@@ -44,7 +44,7 @@ class Node;
 
 namespace interpreter {
 class ViewContext;
-class RelationWrapper;
+struct RelationWrapper;
```
Comment (mmcgr) [inline src/interpreter/Node.h:47]: This change seems unrelated to this PR. It would be much better to put unrelated changes into another PR. That said, I think a better code change would be to leave this as `class` (an improvement in readability over having to think about why `struct` was used) and change the class itself to use `class` rather than `struct`.

Comment (b-scholz) [reply src/interpreter/Node.h]: I fixed this but could we avoid forward references.

**Review by mmcgr — APPROVED** [review]
